src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h

```diff
@@ -171,13 +207,14 @@ inline void postcomputeImposedStateFromNewStress(
     real imposedStateMinus[tensor::QInterpolated::size()],
     const real qInterpolatedPlus[CONVERGENCE_ORDER][tensor::QInterpolated::size()],
     const real qInterpolatedMinus[CONVERGENCE_ORDER][tensor::QInterpolated::size()],
-    const double timeWeights[CONVERGENCE_ORDER]) {
+    const double timeWeights[CONVERGENCE_ORDER],
+    unsigned startIndex = 0) {
 
   // set imposed state to zero
-#ifdef ACL_DEVICE_OFFLOAD
-#pragma omp loop bind(parallel)
-#endif
-  for (unsigned int i = 0; i < tensor::QInterpolated::size(); i++) {
+  using QInterpolatedRange = typename QInterpolated<Type>::Range;
+  for (auto index = QInterpolatedRange::start; index < QInterpolatedRange::end;
+       index += QInterpolatedRange::step) {
+    auto i{startIndex + index};
```

Comment:
Also consider to put this into the loop header.

Reply:
discussed